src/Uno.UI.Runtime.Skia.X11/LinuxSystemThemeHelper.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Threading.Tasks;
+using Tmds.DBus.Protocol;
+using Uno.Foundation.Logging;
+using Uno.Helpers.Theming;
+using Uno.UI.Dispatching;
+using Uno.WinUI.Runtime.Skia.X11.DBus;
+
+namespace Uno.WinUI.Runtime.Skia.X11;
+
+// https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Settings.html
+
+internal class LinuxSystemThemeHelper : ISystemThemeHelperExtension
+{
+	private const string Service = "org.freedesktop.portal.Desktop";
+	private const string ObjectPath = "/org/freedesktop/portal/desktop";
+
+	public event EventHandler? SystemThemeChanged;
+
+	private SystemTheme _currentTheme = SystemTheme.Light;
+	private SystemTheme CurrentTheme
+	{
+		get => _currentTheme;
+		set
+		{
+			if (NativeDispatcher.Main.HasThreadAccess)
+			{
+				if (_currentTheme != value)
+				{
+					_currentTheme = value;
+					SystemThemeChanged?.Invoke(this, EventArgs.Empty);
+				}
+			}
+			else
+			{
+				NativeDispatcher.Main.Enqueue(() => CurrentTheme = value);
+			}
+		}
+	}
+
+	public SystemTheme GetSystemTheme() => CurrentTheme;
+
+	public static LinuxSystemThemeHelper Instance { get; } = new();
+
+	private LinuxSystemThemeHelper()
+	{
+		_ = Init().ConfigureAwait(false);
+	}
+
+	public async Task Init()
+	{
+		try
+		{
+			var sessionsAddressBus = Address.Session;
+			if (sessionsAddressBus is null)
+			{
+				if (this.Log().IsEnabled(LogLevel.Error))
+				{
+					this.Log().Error($"Can not determine DBus session bus address");
+				}
+				return;
+			}
+
+			var connection = new Connection(sessionsAddressBus);
+			await connection.ConnectAsync();
+
+			var desktopService = new DesktopService(connection, Service);
+			var settings = desktopService.CreateSettings(ObjectPath);
+
+			var version = await settings.GetVersionAsync();
+			if (version != 2)
+			{
+				if (this.Log().IsEnabled(LogLevel.Error))
+				{
+					this.Log().Error($"File pickers are only implemented for version 2 of the Settings portal, but version {version} was found");
```

Comment:
Is this message expected to be about file pickers? Otherwise, mention system theme.

Reply:
Oops :)